waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -814,6 +814,12 @@
   [{:keys [replicaset-spec-builder-fn] :as scheduler} service-id service-description context]
   (replicaset-spec-builder-fn scheduler service-id service-description context))
 
+(defn- server-error?
+  "Checks if the given exception object corresponds to a 5XX response indicating a server error."
+  [maybe-exception]
+  (when-let [status (-> maybe-exception ex-data :status)]
+    (<= 500 status 599)))
```

Comment:
Can we use `status-codes.http-500-internal-server-error`?
Nitpick: can we move this function to http-utils.clj?